### PR TITLE
Work in progress on smart contract optimizations

### PR DESCRIPTION
**Pull request by Brando753**

### CODE REVIEW

File: smart-contracts/test/test_bridgeBankLock.js [hunk 1]
```diff
@@ -210,7 +210,7 @@ describe("Test Bridge Bank", function () {
           [false, false, false]
         );
 
-      (await state.bridgeBank.connect(userOne).lockBurnNonce()).should.be.bignumber.equal(previousNonce.add(3));
+      (await state.bridgeBank.connect(userOne).lockBurnNonce()).should.be.equal(previousNonce.add(3));
```
Comment (Brando753) [inline smart-contracts/test/test_bridgeBankLock.js:213]: Why is this change necessary, shouldn't this be a bignumber?

File: smart-contracts/contracts/CosmosBridge.sol [hunk 2]
```diff
@@ -204,7 +180,8 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
     view
     returns (uint256 pow)
   {
```
Comment (smartyalgo) [inline smart-contracts/contracts/CosmosBridge.sol:190]: I couldn't find any unit tests for this function?

File: smart-contracts/contracts/BridgeBank/BridgeBank.sol [hunk 3]
```diff
@@ -542,29 +544,31 @@ contract BridgeBank is BankStorage, CosmosBank, EthereumWhiteList, CosmosWhiteLi
...
+    require(tokenLength == amount.length, "M_P");
+    require(tokenLength == isBurn.length, "M_P");
 
-    uint256 recipientLength = recipient.length;
 
     // lockBurnNonce contains the previous nonce that was
     // sent in the LogLock/LogBurn, so the first one we send
     // should be lockBurnNonce + 1
-    uint256 startingLockBurnNonce = lockBurnNonce + 1;
+    uint256 startingLockBurnNonce = lockBurnNonce + 1;
```
Comment (smartyalgo) [inline smart-contracts/contracts/BridgeBank/BridgeBank.sol:559]: Unnecessary trailing white space

**Review by smartyalgo — CHANGES_REQUESTED** [review]

**Comment by smartyalgo** [comment]
Else LGTM

**Comment by smartyalgo** [comment]
Check if build is successful. GitHub action Smart contract/test should've succeeded

File: smart-contracts/contracts/CosmosBridge.sol (hunk 2 above)
Comment (Brando753) [reply smart-contracts/contracts/CosmosBridge.sol:190]: Its private so we can't call it from the test scripts, it gets tested when batchSubmitProphecyClaimAggregatedSigs is tested

Comment (smartyalgo) [reply smart-contracts/contracts/CosmosBridge.sol:190]: I don't think theres tests to verify duplicate signers would trigger revert

Comment (Brando753) [reply smart-contracts/contracts/CosmosBridge.sol:190]: you available on slack to huddle?

Comment (Brando753) [reply smart-contracts/contracts/CosmosBridge.sol:190]: See testSignatureAggregationSecurity.js
```js
    it("there are duplicate signers", async function () {
      state.recipient = userOne;
      state.nonce = 1;

      const { digest, claimData, signatures } = await getValidClaim({
        sender: state.sender,
        senderSequence: state.senderSequence,
        recipientAddress: state.recipient.address,
        tokenAddress: state.troll.address,
        amount: state.amount,
        doublePeg: false,
        nonce: state.nonce,
        networkDescriptor: state.networkDescriptor,
        tokenName: state.name,
        tokenSymbol: state.symbol,
        tokenDecimals: state.decimals,
        cosmosDenom: state.constants.denom.none,
        validators: [userOne, userTwo, userFour, userFour],
      });

      await expect(
        state.cosmosBridge
          .connect(userOne)
          .submitProphecyClaimAggregatedSigs(digest, claimData, signatures)
      ).to.be.revertedWith("DUP_SIGNER");
    });
    ```

File: smart-contracts/Makefile.artifacts [hunk 4]
```diff
@@ -1 +1 @@
...
...wan.sol/Rowan.json artifacts/contracts/BridgeBank/BridgeToken.sol/BridgeToken.json artifacts/contracts/BridgeBank/EthereumWhitelist.sol/EthereumWhiteList.json artifacts/contracts/BridgeBank/BridgeBank.sol/BridgeBank.json artifacts/contracts/BridgeBank/Pausable.sol/Pausable.json artifacts/contracts/BridgeBank/EthereumBankStorage.sol/EthereumBankStorage.json artifacts/contracts/BridgeBank/PauserRole.sol/PauserRole.json artifacts/contracts/BridgeBank/CosmosBankStorage.sol/CosmosBankStorage.json artifacts/contracts/OracleStorage.sol/OracleStorage.json artifacts/contracts/interfaces/IBlocklist.sol/IBlocklist.json artifacts/contracts/interfaces/IBridgeToken.sol/IBridgeToken.json artifacts/contracts/Oracle.sol/Oracle.json artifacts/contracts/CosmosBridge.sol/CosmosBridge.json artifacts/contracts/Mocks/FailHardToken.sol/FailHardToken.json artifacts/contracts/Mocks/Erowan.sol/Erowan.json artifacts/contracts/Mocks/ReentrancyToken.sol/ReentrancyToken.json artifacts/contracts/Mocks/FakeERC20.sol/FakeERC20.json artifacts/contracts/Mocks/ManyDecimalsToken.sol/ManyDecimalsToken.json artifacts/contracts/Mocks/TrollToken.sol/TrollToken.json artifacts/contracts/BridgeRegistry.sol/BridgeRegistry.json
```
Comment (smartyalgo) [inline smart-contracts/Makefile.artifacts:1]: Should these be committed?

File: smart-contracts/Makefile.smartContracts [hunk 5]
```diff
@@ -1 +1 @@
...
+smartContracts=contracts/Blocklist.sol contracts/Valset.sol contracts/ValsetStorage.sol contracts/CosmosBridgeStorage.sol contracts/BridgeBank/CosmosWhiteList.sol contracts/BridgeBank/CosmosWhiteListStorage.sol contracts/BridgeBank/BankStorage.sol contracts/BridgeBank/CosmosBank.sol contracts/BridgeBank/Rowan.sol contracts/BridgeBank/BridgeToken.sol contracts/BridgeBank/EthereumWhitelist.sol contracts/BridgeBank/BridgeBank.sol contracts/BridgeBank/Pausable.sol contracts/BridgeBank/EthereumBankStorage.sol contracts/BridgeBank/PauserRole.sol contracts/BridgeBank/CosmosBankStorage.sol contracts/OracleStorage.sol contracts/interfaces/IBlocklist.sol contracts/interfaces/IBridgeToken.sol contracts/Oracle.sol contracts/CosmosBridge.sol contracts/Mocks/FailHardToken.sol contracts/Mocks/Erowan.sol contracts/Mocks/ReentrancyToken.sol contracts/Mocks/FakeERC20.sol contracts/Mocks/ManyDecimalsToken.sol contracts/Mocks/TrollToken.sol contracts/BridgeRegistry.sol
```
Comment (smartyalgo) [inline smart-contracts/Makefile.smartContracts:1]: Should these be committed?

Comment (Brando753) [reply smart-contracts/Makefile.smartContracts:1]: Yup, we no longer need migrations since moving from truffle.

**Review by smartyalgo — APPROVED** [review]
